### PR TITLE
add unit test greet

### PR DESCRIPTION
**Pull request by PulexIrritans**

### CODE REVIEW

**Review by ThomasF85 — APPROVED** [review]
Schöne Logik!
Auch dass du hier mit "early returns" arbeitest anstatt mit if() {} else if() {} else if() ... ist schön!

File: greet.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export function greet(name) {
+    
+    if (name === 'Thomas' || name === 'Martin' || name === 'Stefan') return 'Hello Coach!'
+    if (name === '') return 'Hello stranger!' 
+    if (name === undefined) return 'Hello stranger!'
```
Comment (mpagels) [inline greet.js]: Awesome using the short version of `if/else`

Comment (mpagels) [inline greet.js]: You could combine these two like line 3
```suggestion
    if (name === '' || name === undefined) return 'Hello stranger!' 
```

File: greet.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    else return `Hello ${name}!`
```
Comment (mpagels) [inline greet.js:5]: Slick, using `template literals` here. ❤️

File: greet.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export function greet(name) {
```
Comment (mpagels) [inline greet.js:1]: You could use here a `default parameters` to be sure not have an `undefined` argument.
```suggestion
export function greet(name="") {
```

File: greet.spec.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    })
+    it('return the greeting for Thomas is `Hello Coach!`', () => {
+        const result = greet('Thomas')
+        expect(result).toBe('Hello Coach!')
+    })
+    it('return the greeting for Antje is `Hello Antje!`', () => {
+        const result = greet('Antje')
+        expect(result).toBe('Hello Antje!')
+    })
+})
```
Comment (mpagels) [inline greet.spec.js:20]: Awesome. Perfect.

**Review by mpagels — APPROVED** [review]
Awesome pr. Have some tiny things. Just tips. Pls check it out!

One big recommendation: Use the `pull request description`!. Here you should provide as much information to the reviewer as possible. When the reviewers take the time to review your code, try to guide them. It doesn't matter how small the `pull request` is. Here you could write:

> This pr adds a test for the `greet` function to the project.

Now the reviewer knows, on what to focus.

Again, awesome pull request!

**Pull request merged**